x-pack/plugins/security_solution/server/lib/entity_analytics/utils/transforms.ts

```diff
@@ -119,50 +124,74 @@ export const startTransform = async ({
   });
   if (transformStats.count <= 0) {
     throw new Error(
-      `Unable to find transform status for [${transformId}] while attempting to start`
+      `Unable to find transform status for [${transformId}] while attempting to schedule`
     );
   }
-  if (hasTransformStarted(transformStats.transforms[0])) {
-    return;
-  }
 
-  return esClient.transform.startTransform({ transform_id: transformId });
+  if (!hasTransformStarted(transformStats.transforms[0])) {
+    await esClient.transform.startTransform({
+      transform_id: transformId,
+    });
+  } else {
+    await esClient.transform.scheduleNowTransform({
+      transform_id: transformId,
+    });
+  }
 };
 
-export const scheduleTransformNow = async ({
+const upgradeLatestTransformIfNeeded = async ({
   esClient,
-  transformId,
+  namespace,
+  logger,
 }: {
   esClient: ElasticsearchClient;
-  transformId: string;
+  namespace: string;
+  logger: Logger;
 }): Promise<TransformStartTransformResponse | void> => {
-  const transformStats = await esClient.transform.getTransformStats({
+  const transformId = getLatestTransformId(namespace);
+  const latestIndex = getRiskScoreLatestIndex(namespace);
+  const timeSeriesIndex = getRiskScoreTimeSeriesIndex(namespace);
+
+  const response = await esClient.transform.getTransform({
     transform_id: transformId,
   });
-  if (transformStats.count <= 0) {
-    throw new Error(
-      `Unable to find transform status for [${transformId}] while attempting to schedule now`
-    );
-  }
 
-  if (hasTransformStarted(transformStats.transforms[0])) {
-    await esClient.transform.scheduleNowTransform({
-      transform_id: transformId,
+  if (
+    response.transforms[0].sync?.time?.delay === '2s' ||
```

Comment:
Instead of making this explicit, could we reverse this and pull from the configuration to see what value is desired?

`response.transforms[0].sync?.time?.delay !== getTransformOptions(dest, source).sync.time.delay`

Reply:
You gave me an interesting idea... We can compare the entire object and make it future-proof. 
`if (isEqual(newConfig, oldConfig)) {`

Reply:
Aborting... The stored transform has more properties.

Reply:
I created a whitelist. What do you think @jaredburgettelastic?

Reply:
I like the field whitelist approach, and the `isTransformOutdated` function is a nice abstraction as well 👍